sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -498,7 +498,8 @@ class Dataset[T] private[sql](
    * @since 1.6.0
    */
   def explain(extended: Boolean): Unit = {
-    val explain = ExplainCommand(queryExecution.logical, extended = extended)
+    val explain = ExplainCommand(queryExecution.logical, extended = extended,
```

Comment:
what about passing the `QueryExecution` as first parameter? IIUC, there is only another place where `ExplainCommand` is created, so it is not going to be a big change too and it is going to be cleaner IMO...

Reply:
`ExplainCommand` is also created in `SparkSqlParser` where there is no `QueryExecution`. So it has both `LogicalPlan` and `QueryExecution` parameters.

Reply:
Yes, my point is: can we create a `QueryExecution` in `SparkSqlParser` and pass that to `ExplainCommand`?

Reply:
Made a change. Please see if it is clearer for you.